storage/pruning/pruningStorer.go

```diff
@@ -0,0 +1,414 @@
+package pruning
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"reflect"
+	"regexp"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/epochStart/notifier"
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
+)
+
+var log = logger.GetOrCreate("storage/pruning")
+
+// DefaultEpochDirectoryName represents the naming pattern for epoch directories
+const DefaultEpochDirectoryName = "Epoch"
+
+// persisterData structure is used so the persister and it's path can be kept in the same place
+type persisterData struct {
+	persister storage.Persister
+	path      string
+}
+
+// PruningStorer represents a storer which creates a new persister for each epoch and removes older persisters
+type PruningStorer struct {
+	lock                  sync.RWMutex
+	fullArchive           bool
+	persisters            []*persisterData
+	closedPersistersPaths []string
+	cacher                storage.Cacher
+	bloomFilter           storage.BloomFilter
+	dbPath                string
+	persisterFactory      DbFactoryHandler
+	numOfEpochsToKeep     uint32
+	numOfActivePersisters uint32
+	identifier            string
+}
+
+// NewPruningStorer will return a new instance of PruningStorer without sharded directories' naming scheme
+func NewPruningStorer(args *PruningStorerArgs) (*PruningStorer, error) {
+	return initPruningStorer(args, "")
+}
+
+// NewShardedPruningStorer will return a new instance of PruningStorer with sharded directories' naming scheme
+func NewShardedPruningStorer(
+	args *PruningStorerArgs,
+	shardId uint32,
+) (*PruningStorer, error) {
+	shardIdStr := fmt.Sprintf("%d", shardId)
+	return initPruningStorer(args, shardIdStr)
+}
+
+// initPruningStorer will create a PruningStorer with or without sharded directories' naming scheme
+func initPruningStorer(
+	args *PruningStorerArgs,
+	shardIdStr string,
+) (*PruningStorer, error) {
+	var cache storage.Cacher
+	var db storage.Persister
+	var bf storage.BloomFilter
+	var err error
+
+	defer func() {
+		if err != nil && db != nil {
+			_ = db.Destroy()
+		}
+	}()
+
+	if args.NumOfActivePersisters < 1 {
+		return nil, storage.ErrInvalidNumberOfPersisters
+	}
+	if check.IfNil(args.Notifier) {
+		return nil, storage.ErrNilEpochStartNotifier
+	}
+	if check.IfNil(args.PersisterFactory) {
+		return nil, storage.ErrNilPersisterFactory
+	}
+
+	cache, err = storageUnit.NewCache(args.CacheConf.Type, args.CacheConf.Size, args.CacheConf.Shards)
+	if err != nil {
+		return nil, err
+	}
+
+	filePath := args.DbPath
+	if len(shardIdStr) > 0 {
+		filePath = filePath + shardIdStr
+	}
+	db, err = args.PersisterFactory.Create(filePath)
+
+	var persisters []*persisterData
+	persisters = append(persisters, &persisterData{
+		persister: db,
+		path:      filePath,
+	})
+
+	if reflect.DeepEqual(args.BloomFilterConf, storageUnit.BloomConfig{}) {
```

Comment:
verify only one parameter from the config to decide whether you want with bloom config or not.

Reply:
done. checked for bloom's size. if the struct is empty (so no config for bloom filter), the default value will be 0